internal/core/src/storage/DiskFileManagerImpl.cpp

```diff
@@ -137,20 +138,30 @@ DiskFileManagerImpl::CacheIndexToDisk(std::vector<std::string> remote_files) {
         auto prefix = slices.first;
         auto local_index_file_name = GetLocalIndexObjectPrefix() + prefix.substr(prefix.find_last_of("/") + 1);
         local_chunk_manager.CreateFile(local_index_file_name);
-        int64_t offset = 0;
+
+        std::vector<std::string> files;
+        std::map<std::string, uint64_t> file_to_size;
         for (auto iter = slices.second.begin(); iter != slices.second.end(); iter++) {
             auto origin_file = prefix + "_" + std::to_string(*iter);
+            files.emplace_back(origin_file);
             auto fileSize = remote_chunk_manager.Size(origin_file);
+            file_to_size[origin_file] = fileSize;
+        }
+
+        auto batch_size = milvus::config::KnowhereGetIndexSliceSize() << 20;
+#pragma omp parallel for
+        for (auto i = 0; i < files.size(); i++) {
+            auto origin_file = files[i];
```

Comment:
variable origin_file is still in use ?

Reply:
> is variable origin_file still in use ?

origin_file is used to get file size from file_to_size(std::map<std::string, uint64_t>)